parser/src/main/java/com/facebook/coresql/sqllogictest/SqlLogicTest.java

```diff
@@ -0,0 +1,308 @@
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.text.DecimalFormat;
+import java.util.HashSet;
+import java.util.Properties;
+import java.util.Set;
+
+import static com.facebook.coresql.parser.ParserHelper.parseStatement;
+import static com.facebook.coresql.parser.Unparser.unparse;
+import static java.lang.String.format;
+import static java.nio.file.FileVisitResult.CONTINUE;
+
+public final class SqlLogicTest
+{
+    private static AcceptedRdms currRdms;
+    private static Set<AcceptedRdms> skipIfSet;
+    private static AcceptedRdms onlyIf;
+    private static BufferedWriter writer;
+    private static FileVisitor<Path> fileVisitor;
+    private static String startingDir;
+    private static String resultsDir;
+    private static int maxFilesToVisit;
+    private static int numFilesVisited;
+    private static int numStatementsParsed;
+    private static int numStatementsFound;
+
+    private enum SqlLogicTestError
+    {
+        PARSING_ERROR,
+        UNPARSING_ERROR,
+        UNPARSED_DOES_NOT_MATCH_ERROR
+    }
+
+    private enum AcceptedRdms
+    {
+        ALL,
+        ORACLE,
+        MYSQL,
+        MSSQL,
+        SQLITE,
+        POSTGRESQL
+    }
+
+    private SqlLogicTest() {}
+
+    public static void run()
+            throws IOException
+    {
+        setup();
+        Files.walkFileTree(Paths.get(startingDir), fileVisitor);
+        shutdown();
+    }
+
+    private static void setup()
+            throws IOException
+    {
+        getDataFromConfig();
+        createFileForResults();
+        writer = new BufferedWriter(new FileWriter(resultsDir + "sqllogictest_results.txt"));
+        fileVisitor = new SqlLogicTestFileVisitor();
+        skipIfSet = new HashSet<>();
+        onlyIf = null;
+    }
+
+    private static void shutdown()
+            throws IOException
+    {
+        writer.write(
+                "===============================================\n" +
+                        "RESULTS\n" +
+                        "===============================================\n");
+        writer.write(format("RDMS tested against: %s\n", currRdms));
+        writer.write(format("%d statements found, %d statements successfully parsed.\n", numStatementsFound, numStatementsParsed));
+        writer.write(format("%s%% of sqllogictest statements were successfully parsed.",
+                new DecimalFormat("#.##").format(numStatementsParsed / (double) numStatementsFound * 100)));
+        writer.close();
+    }
+
+    private static void getDataFromConfig()
+            throws IOException
+    {
+        try (InputStream input = new FileInputStream("parser/src/main/resources/config.properties")) {
+            Properties prop = new Properties();
+            prop.load(input);
+            verifyAndSetRdms(prop.getProperty("rdms"));
```

Comment:
You might want to take a look at how presto codebase supports config files and use the same framework here.

Reply:
Just implemented a framework based on the "PrestoFacebookSpark" config code.

Done.